providers/tcp.js

```diff
@@ -20,46 +20,47 @@ var net       = require('net')
 ;
 
 function TcpStream(options) {
-  if(!(this instanceof TcpStream)) {
+  if (!(this instanceof TcpStream)) {
     return new TcpStream(options);
   }
 
   Transform.call(this, options);
 
   this.options    = options;
-  this.reconnect  = options.reconnect || true;
+  this.reconnect  = (typeof options.reconnect === 'boolean' && options.reconnect === false) ? false : true;
   this.socket     = null;
   this.retries    = 0;
-  this.maxRetries = options.maxRetries || 10;
+  this.maxRetries = (typeof options.maxRetries === 'number' && options.maxRetries > 0) ? options.maxRetries : 10;
 
   this.__reset    = null;
   this.__timeout  = null;
   this.__last     = -1;
 
   this.start();
 
-  this.on('error', function(err) {
-    debug('TcpStream', err);
+  this.on('error', function (err) {
+    debug('Stream: "error". Message: ' + err.message);
   });
 }
 
 require('util').inherits(TcpStream, Transform);
 
-TcpStream.prototype.handleTimeout = function() {
-  if((Date.now() - this.__last) > 90000) {
+TcpStream.prototype.handleTimeout = function () {
+  if ((Date.now() - this.__last) > 90000 && this.__reset === null) {
```

Comment:
Won't this cause reconnects when tcp connection is ok but there is no data coming over the connection?